File: rest-service/src/main/java/mb/lib/blast/model/tn/TBlastNTask.java
package mb.lib.blast.model.tn;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import mb.api.service.util.ErrorText;

public enum TBlastNTask
{
  TBlastN("tblastn"),
  TBlastNFast("tblastn-fast");

  private static final TBlastNTask def = TBlastN;

  private final String value;

  TBlastNTask(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  public static TBlastNTask getDefault() {
    return def;
  }

  @Override
  @JsonValue
  public String toString() {
    return value;
  }

  @JsonCreator
  public static TBlastNTask fromString(String value) {
    for (var e : values())
      if (e.value.equals(value))
        return e;

    throw new IllegalArgumentException(String.format(
      ErrorText.InvalidEnumValue,
      value,
      TBlastNTask.class.getSimpleName()
    ));
  }
}

File: lib-jvm-mblast-sdk/src/main/kotlin/org/veupathdb/lib/mblast/sdk/query/blast/TBlastNTask.kt
package org.veupathdb.lib.mblast.sdk.query.blast

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.node.TextNode
import org.veupathdb.lib.mblast.sdk.util.UnrecognizedEnumException
import org.veupathdb.lib.mblast.sdk.util.dEnumRequireTextual

enum class TBlastNTask {
  TBlastN,
  TBlastNFast,
  ;

  @JsonValue
  fun toJson(): JsonNode = TextNode(
    when (this) {
      TBlastN     -> "tblastn"
      TBlastNFast -> "tblastn-fast"
    }
  )

  companion object {

    @JvmStatic
    @JsonCreator
    fun fromJson(js: JsonNode): TBlastNTask {
      js.dEnumRequireTextual(TBlastNTask::class)
      return when (val v = js.textValue()) {
        "tblastn"      -> TBlastN
        "tblastn-fast" -> TBlastNFast
        else           -> throw UnrecognizedEnumException(TBlastNTask::class, v)
      }
    }
  }
}